src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,446 @@
+        {
+            foreach (var platformSpecificOperation in platformSpecificOperations)
+            {
+                ReportDiagnostics(platformSpecificOperation.Key, platformSpecificOperation.Value.FirstOrDefault(), context);
+            }
+        }
+
+        private static void ReportDiagnostics(IOperation operation, PlatformAttributeInfo attribute, OperationBlockAnalysisContext context)
+        {
+            string operationName = string.Empty;
+
+            if (operation is IInvocationOperation iOperation)
+            {
+                operationName = iOperation.TargetMethod.Name;
+            }
+            else if (operation is IObjectCreationOperation cOperation)
+            {
+                operationName = cOperation.Constructor.Name;
+            }
+            else if (operation is IPropertyReferenceOperation pOperation)
+            {
+                operationName = pOperation.Property.Name;
+            }
+            else if (operation is IFieldReferenceOperation fOperation)
+            {
+                operationName = fOperation.Field.Name;
+            }
+            if (operation is IDelegateCreationOperation dOperation)
+            {
+                if (dOperation.Target is IMethodReferenceOperation mrOperation)
+                    operationName = mrOperation.Method.Name;
+            }
+            else if (operation is IEventReferenceOperation eOperation)
+            {
+                operationName = eOperation.Event.Name;
+            }
+
+            context.ReportDiagnostic(operation.CreateDiagnostic(SelectRule(attribute.AttributeType),
+                operationName, attribute.PlatformName, attribute.Version.ToString()));
+        }
+
+        private static void AnalyzeInvocationOperation(IOperation operation, OperationAnalysisContext context,
+            ref PooledDictionary<IOperation, ImmutableArray<PlatformAttributeInfo>> platformSpecificOperations)
+        {
+            using var builder = ArrayBuilder<PlatformAttributeInfo>.GetInstance();
+            AttributeData[]? attributes = null;
+
+            if (operation is IInvocationOperation iOperation)
+            {
+                attributes = FindAllPlatformAttributesApplied(iOperation.TargetMethod.GetAttributes(), iOperation.TargetMethod.ContainingType);
+            }
+            else if (operation is IObjectCreationOperation cOperation)
+            {
+                attributes = FindAllPlatformAttributesApplied(cOperation.Constructor.GetAttributes(), cOperation.Constructor.ContainingType);
+            }
+            else if (operation is IPropertyReferenceOperation pOperation)
+            {
+                attributes = FindAllPlatformAttributesApplied(pOperation.Property.GetAttributes(), pOperation.Property.ContainingType);
+            }
+            else if (operation is IFieldReferenceOperation fOperation)
+            {
+                if (!IsWithinConditionalOperation(operation))
```

Comment:
Why is this required? This indicates likely false positives in the analyzer

Reply:
This is actually for reducing false positives, the requirement is:
`if (value != FooEnum.WindowsOnlyValue)`  // should not cause a warning, but:
`FooMethod(FooEnum.WindowsOnlyValue)`  // should warn

Reply:
Interesting, can you please add a comment on the same lines. This means we will report diagnostics if the field is saved into a local and then checked later by comparing the local? I wonder if a better approach is to only flag field references that are passed as arguments to methods that are platform specific, so analysis happens in context of the invoked method’s platform information instead of the field.